docs/rfcs/XeGPU.md

```diff
@@ -286,7 +245,236 @@ In case that certain Xe GPU target does not support atomic operation for a certa
+```
+
+Size distribution rule can be represented as following: tensor_desc[0] must be divisible by wi_layout[0] x wi_data[0], tensor_desc[1] by wi_layout[1] x wi_data[1], and tensor_desc_size by sg_map_size. The 2D subtensor is evenly distributed to WI threads, so each WI thread gets a 2D data fragments. 
+
+The size of the result data fragement per WI thread can be computed by the following: 
+```mlir
+	WI_data_frag[0] = tensor_desc[0]/wi_layout[0]
+	WI_data_frag[1] = tensor_desc[1]/wi_layout[1]
+```
+
+The WI dsitribution is represented by the shape of the result vector being loaded, which is being reduced from [tensor_decs[0], tensor_desc[1]] to [WI_data_frag[0], WI_data_frag[1]].
+
+With `sg_map` attribute attached to tensor_desc, XeGPU.load_nd operates in SIMT flavor and returns back a fragement associated with individual WI thread. The tensor_desc in the example below specifies a tensor of 8x16 elements, which is decomposed to 8x1 subtensors, each with sg_map_size 1x16. The result vector <8x16xbf16> is loaded and distributed to each WI as <8x1xbf16>.  
+```mlir
+  #sg_map_a = xegpu.sg_map<wi_layout = [1, 16], wi_data = [1, 1]>
+  %vector_a = XeGPU.load_nd %tdesc_a:
+     tensor_desc<8x16xbf16, #sg_map_b> into vector<8x1xbf16>
+```
+
+For load_nd with `packed` attribute, wi_data[0] must equal to the size required to “pack” the low-precision data into 32-bit, also known as `vnni_factor`. The result vector takes wi_data[0] as inner dimension size, to indicate the effects of layout change known as "VNNI" transformation. The data fragement <16x1xbf16> is loaded and packed as <8x1x2xbf16>. 
+
+```mlir
+  #sg_map_b = xegpu.sg_map<wi_layout = [1, 16], wi_data = [2, 1]>
+  %vector_b = XeGPU.load_nd {packed} %tdesc1:
+     tensor_desc<16x16xbf16, #sg_map_b> into vector<8x1x2xbf16>
+```
+
+For load_nd with `transpose` attribute, wi_layout is transposed to match with the tensor dimension swap. The tensor is first distributed to WI using `sg_map`, so each WI get 1x8xf32 in the example below, and then transposed to 8x1xf32. The data fragement <8x1xf32> is loaded and transposed as <1x8xf32>.
+```mlir
+  #sg_map_at = xegpu.sg_map<wi_layout = [16, 1], wi_data = [1, 1]>
+  %at = XeGPU.load_nd %tdesc1 {transpose = [1,0]} :
+     tensor_desc<16x8xf32, #sg_map_at> into vector<8x1xf32>
+```
+`xegpu.sg_map` is also applied to 1d vector load for WI data distribution. When the tensor_desc only specify 1d tensor, `sg_map.wi_layout[1]` and `sg_map.wi_data[0]` must be 1, and they are ignored in the WI distribution. 
+
+`xegpu.sg_map` is also used to describe the WI data distribution for regular load. Below example shows that each WI loads one fp32 data element. The result vector <16xfp32> is loaded and distributed to each WI as <1xf32>.
+```mlir
+  #sg_map_t = xegpu.sg_map<wi_layout = [1, 16], wi_data = [1, 1]>
+  #scatter_attr = !xegpu.tdesc_attr< memory_scope=slm, scattered=true>
+  %scatter_tdesc = XeGPU.create_tdesc, %src_addr, %offsets:
```

Comment:
let's use lower register across all examples so change XeGPU->xegpu everywhere in code snippets